tests/core/test_featurizer.py

```diff
@@ -36,29 +37,76 @@ def test_fail_to_load_non_existent_featurizer():
     assert TrackerFeaturizer.load("non_existent_class") is None
 
 
-def test_binary_featurizer_correctly_encodes_state():
-    f = BinarySingleStateFeaturizer()
+def test_single_state_featurizer_correctly_encodes_state():
+    """
+    Check that all the attributes are correctly featurized when they should and not featurized when shouldn't;
+    These tests are for encoding state without a trained interpreter.
+    """
+    f = SingleStateFeaturizer()
     f._default_feature_states[INTENT] = {"a": 0, "b": 1}
-    f._default_feature_states[ACTION_NAME] = {"c": 0, "d": 1}
+    f._default_feature_states[ACTION_NAME] = {"c": 0, "d": 1, "action_listen": 2}
+    f._default_feature_states[SLOTS] = {"e_0": 0, "f_0": 1, "g_0": 2}
+    f._default_feature_states[ACTIVE_LOOP] = {"h": 0, "i": 1, "j": 2, "k": 3}
+
     encoded = f.encode_state(
-        {"user": {"intent": "a"}, "prev_action": {"action_name": "d"}}, interpreter=None
+        {
+            "user": {"intent": "a", "text": "blah blah blah"},
+            "prev_action": {"action_name": "d", "action_text": "boom"},
+            "active_loop": {"name": "i"},
+            "slots": {"g": (1.0,)},
+        },
+        interpreter=None,
     )
-    # user input is ignored as prev action is not action_listen;
-    assert list(encoded.keys()) == [ACTION_NAME]
+    # user input is ignored as prev action is not action_listen
+    assert list(encoded.keys()) == [ACTION_NAME, ACTIVE_LOOP, SLOTS]
+    assert (
+        encoded[ACTION_NAME][0].features != scipy.sparse.coo_matrix([[0, 1, 0]])
+    ).nnz == 0
     assert (
-        encoded[ACTION_NAME][0].features != scipy.sparse.coo_matrix([[0, 1]])
+        encoded[ACTIVE_LOOP][0].features != scipy.sparse.coo_matrix([[0, 1, 0, 0]])
     ).nnz == 0
+    assert (encoded[SLOTS][0].features != scipy.sparse.coo_matrix([[0, 0, 1]])).nnz == 0
 
     encoded = f.encode_state(
-        {"user": {"intent": "a"}, "prev_action": {"action_name": "action_listen"}},
+        {
+            "user": {"intent": "a", "text": "blah blah blah"},
+            "prev_action": {"action_name": "action_listen", "action_text": "boom"},
+            "active_loop": {"name": "k"},
+            "slots": {"e": (1.0,)},
+        },
         interpreter=None,
     )
-    assert list(encoded.keys()) == [INTENT, ACTION_NAME]
+    # we featurize all the features except for *_text ones because NLU wasn't trained
```

Comment:
let's put that also in the name of the test `..._without_interpreter`

Reply:
done!